### PR TITLE
Issue 1842: Use JPA and include order by on vocabulary when ordering Controlled Vocabulary dictionary and revert input select.

### PR DESCRIPTION
**Pull request by kaladay**
Relates #1851

Use JPA instead of Hibernate to perform the OrderBy. The Hibernate uses SQL and JPA is the preferred approach.

The Degree information still needs sorting, add back orderBy for input select.

At least if most of the information is sorted already, then there will still be a performance boost of not having to move anything around in the case of the input select.

### CODE REVIEW

File: src/main/webapp/app/views/inputtype/input-select.html [hunk 1]
```diff
@@ -9,7 +9,7 @@
     <li class="select-option"
       role="menuitem"
       ng-click="fieldValue.value = word.name; fieldProfileForm.$dirty = true; saveWithCV(fieldValue, word);"
-      ng-repeat="word in profile.controlledVocabulary.dictionary | filter:displayVocabularyWord">
+      ng-repeat="word in profile.controlledVocabulary.dictionary | filter:displayVocabularyWord | orderBy:'name'">
```
Comment (wwelling) [inline src/main/webapp/app/views/inputtype/input-select.html:12]: Are there other locations where `orderBy:'name'` sorting removed recently?

Comment (kaladay) [reply src/main/webapp/app/views/inputtype/input-select.html:12]: See the PR referenced in the comments above.
Linked here for convenience:
- https://github.com/TexasDigitalLibrary/Vireo/pull/1851

Comment (wwelling) [reply src/main/webapp/app/views/inputtype/input-select.html:12]: Yes, thank you. That is the PR I glanced at. That PR changed multiple template loop expressions. Do any of those need to be reverted back? Or just the one?

Comment (kaladay) [reply src/main/webapp/app/views/inputtype/input-select.html:12]: Just this one as this PR states.

**Review by rmathew1011 — APPROVED** [review]

**Pull request merged**